app/controllers/theme_controller.py

```diff
@@ -0,0 +1,158 @@
+        if not hasattr(self, "_supported_themes"):
+            theme_data_folders = self.get_theme_names_from_folder(
+                self.app_info.theme_data_folder
+            )
+            theme_storage_folders = self.get_theme_names_from_folder(
+                self.app_info.theme_storage_folder
+            )
+            self._supported_themes = theme_data_folders | theme_storage_folders
+            logger.info(f"Supported themes retrieved: {self._supported_themes}")
+            logger.info(
+                f"Checking theme folders: {[self.app_info.theme_data_folder, self.app_info.theme_storage_folder]}"
+            )
+        return self._supported_themes
+
+    def get_theme_names_from_folder(self, folder: Path) -> set[str]:
+        """Helper method to get theme names from a specified folder."""
+        supported_themes = set()
+        if folder.exists():
+            for subfolder in folder.iterdir():
+                if subfolder.is_dir():
+                    stylesheet_path = subfolder / "style.qss"
+                    if stylesheet_path.exists():
+                        supported_themes.add(subfolder.name)
+                        logger.info(
+                            f"Found theme with stylesheet in {folder}: {subfolder.name}"
+                        )
+        return supported_themes
+
+    def load_theme(self, theme_name: str) -> Optional[str]:
+        """Load the specified theme."""
+        if theme_name in self.themes:
+            logger.info(f"Loading theme: {theme_name}")
+            stylesheet_path = self.get_theme_stylesheet_path(theme_name)
+            if stylesheet_path:
+                try:
+                    with open(stylesheet_path, "r") as f:
+                        stylesheet = f.read()
+                    return stylesheet
+                except Exception as e:
+                    logger.error(f"Error loading theme: {e}")
+                    return None
+            else:
+                logger.error(f"Stylesheet not found for theme: {theme_name}")
+                return None
+        else:
+            logger.error(f"Attempted to load unsupported theme: {theme_name}")
+            # Revert to the default theme "RimPy"
+            self.theme_name = self.default_theme
+            return self.load_theme(
+                self.default_theme
+            )  # Attempt to load the default theme
+
+    def get_theme_stylesheet_path(self, theme_name: str) -> Optional[Path]:
+        """Returns the path to the stylesheet for the specified theme."""
+        logger.info(f"Searching for stylesheet for theme: {theme_name}")
+
+        # Initialize a variable to hold the path
+        stylesheet_path = None
+
+        for folder in self.theme_stylesheets_folder:
+            # Construct the stylesheet path
+            potential_path = folder / theme_name / "style.qss"
+            logger.debug(f"Checking for stylesheet at: {potential_path}")
```

Comment:
I still don't really agree with this method of storing themes. Whats the reasoning for getting rid of the the middle "theme_name" folder and using the .qss file name as the theme name? That seems to be standard in other QT apps.